### PR TITLE
strip-coverletter-docker, changes work root from /tmp to /bot-tmp/tmp

### PR DESCRIPTION
**Pull request by lsh-0**
but only if /bot-tmp exists.

### CODE REVIEW

File: strip-coverletter-docker.sh [hunk 1]
```diff
@@ -1,24 +1,32 @@
...
+out_dir=$(dirname "$out")
 bname_out=${out##*/}
 
-work_dir="/tmp/decap"
-logfile="$work_dir/$bname_in.log" # ll: /tmp/decap/baz.pdf.log
+# lsh@2022-08-02: default to the external volume mounted at /bot-tmp, if it exists.
+# this is easier than modifying elife-bot code.
+tmp_dir="/tmp"
+if [ -e "/bot-tmp" ]; then
+    tmp_dir="/bot-tmp/tmp"
```
Comment (gnott) [inline strip-coverletter-docker.sh]: Now looking at this after approving the PR in `elife-bot-formula`, I'm not sure the `tmp` folder can be relied on to be here. You may consider just using `/bot-tmp` and then the decap files would presumably end up in `/bot-tmp/decap` folder.

File: strip-coverletter-docker.sh [hunk 2]
```diff
@@ -1,24 +1,32 @@
 #!/bin/bash
 set -eu
 
-in=$(realpath $1)
-bname_in=${in##*/} # /foo/bar/baz.pdf = baz.pdf
+in=$(realpath "$1")
```
Comment (gnott) [inline strip-coverletter-docker.sh:4]: I like the quotation marks good touch 👍

**Review by gnott — APPROVED** [review]
Optionally you can go with the subfolder as you have it or change it, all good

File: strip-coverletter-docker.sh (hunk 2 above)
Comment (lsh-0) [reply strip-coverletter-docker.sh:4]: Thanks to this brilliant tool: https://github.com/koalaman/shellcheck

I've been using it for years but this strip-coverletter script is some of my earliest bash work.

File: strip-coverletter-docker.sh (hunk 1 above)
Comment (lsh-0) [reply strip-coverletter-docker.sh]: that's a good point, and `/bot-tmp/tmp/decap` is convoluted.

**Comment by lsh-0** [comment]
changes made, merging.

let me know if the strip-coverletter process starts misbehaving.

**Pull request merged**